### PR TITLE
DBZ-3292 Prefer archived logs over redo logs for same SCN range.

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-3292

### CODE REVIEW

File: debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/LogMinerHelperIT.java [hunk 1]
```diff
@@ -97,18 +102,24 @@ public void shouldAddRightRedoFiles() throws Exception {
...
             return true;
         }).map(LogFile::getFileName).collect(Collectors.toList());
-        int archivedLogFilesCount = archivedLogFiles.size();
+        int redoLogFilesCount = redoLogFiles.size();
 
-        Map<String, String> redoLogFiles = LogMinerHelper.getMap(conn, SqlUtils.allOnlineLogsQuery(), "-1");
-        assertThat(getNumberOfAddedLogFiles(conn) == (redoLogFiles.size() + archivedLogFilesCount)).isTrue();
+        if (!archivedLogFilesForMining.isEmpty()) {
+            assertThat(onlineLogFilesForMining.size()).isGreaterThan(redoLogFilesCount);
+            assertThat(getNumberOfAddedLogFiles(conn)).isGreaterThan(redoLogFilesCount);
```
[the remark this answers is not in the document]
Comment (Naros) [reply debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/LogMinerHelperIT.java:117]: Are you suggesting we change the following:
```
        if (!archivedLogFilesForMining.isEmpty()) {
            assertThat(onlineLogFilesForMining.size()).isGreaterThan(redoLogFilesCount);
            assertThat(getNumberOfAddedLogFiles(conn)).isGreaterThan(redoLogFilesCount);
        }
        else {
            assertThat(onlineLogFilesForMining.size()).isEqualTo(redoLogFilesCount);
            assertThat(getNumberOfAddedLogFiles(conn)).isEqualTo(redoLogFilesCount);
        }
``` 
to this:
```
        assertThat(redoLogFilesCount + archivedLogFilesForMining.size()).isGreaterThanOrEqualTo(redoLogFilesCount);
        assertThat(getNumberOfAddedLogFiles(conn)).isGreaterThanOrEqualTo(redoLogFilesCount);
```

[the remark this answers is not in the document]
Comment (ckamm) [reply debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/LogMinerHelperIT.java:117]: I was wondering why the test can't compare `redoLogFilesCount + archivedLogFilesForMining.size() == getNumberOfAddedLogFiles(conn)` for both cases. Why would the latter potentially be larger?

[the remark this answers is not in the document]
Comment (Naros) [reply debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/LogMinerHelperIT.java:117]: I follow you now, I'll take a look.

**Review by gunnarmorling — APPROVED** [review]

**Comment by gunnarmorling** [comment]
Thx!

**Pull request merged**